### PR TITLE
Overmap graphics randomness

### PR DESCRIPTION
**Pull request by jbytheway**
#### Summary
Interface "Support randomization between tile variants for overmap graphics"

#### Purpose of change
Overmap graphics can look more natural with some randomization, we want to support it for the same reason we want to support randomization for regular terrain graphics.

#### Describe the solution
Seed randomization with the global omt coordinates of the tile.

This is the natural choice, but was a little tricky.  To make it work I had to change the drawing coordinates of `cata_tiles`.

`cata_tiles` draws relative to an origin `o`.  When drawing the regular map this is a map-local coordinate.  When drawing the overmap this used to be always zero.  After this change it now uses the global OMT coordinates of the north-west corner.  This is helpful because:
* It simplifies `draw_om` which no longer needs to keep track of `omp` and pos separately; they are the same value.
* It allows `pos` within draw_from_id_string to be interpreted meaningfully, in particular to provide the seeding mentioned above.

#### Describe alternatives you've considered
Some further refactoring might be warranted.

#### Testing
As a proof of concept I tweaked RetroDays to provide two random variants of `forest_thick`, using the existing tiles for `forest` and `forest_thick`.  It looks like this:
overmap-randomization

#### Additional context
I know @KorGgenT is currently trying to fix mouse cursor interpretation for clicking on the overmap, and I suspect this change might impact that work.  I'm hoping it might make it easier.

### CODE REVIEW

**Comment by KorGgenT** [comment]
the pr that would affect mouse click has already been merged. we'll see how it all sifts out

File: src/sdltiles.cpp [hunk 1]
```diff
@@ -829,16 +829,15 @@ void cata_tiles::draw_om( const point &dest, const tripoint_abs_omt &center_abs_
                              100;
     const bool showhordes = uistate.overmap_show_hordes;
     const bool viewing_weather = uistate.overmap_debug_weather || uistate.overmap_visible_weather;
-    o = dest;
+    o = corner_NW.raw().xy();
 
     const auto global_omt_to_draw_position = [&corner_NW]( const tripoint_abs_omt & omp ) {
```
Comment (kevingranade) [inline src/sdltiles.cpp:834]:
```suggestion
    const auto global_omt_to_draw_position = []( const tripoint_abs_omt & omp ) {
```

**Comment by int-ua** [comment]
Can you please add an example? It doesn't seem to work with weighted arrays

**Comment by jbytheway** [comment]
Here's an [example being PRed to UltiCa](https://github.com/I-am-Erk/CDDA-Tilesets/pull/734).

**Pull request merged**